modules/serverless-functions-automatic-build-deploy.adoc

```diff
@@ -0,0 +1,92 @@
+// Module included in the following assemblies:
+//
+// * functions/serverless-functions-on-cluster-builds.adoc
+
+:_content-type: PROCEDURE
+[id="serverless-functions-automatic-build-deploy_{context}"]
+= Configuring automatic building and deployment on GitHub Updates
+
+You can use the Shipwright framework to configure a function to be automatically built and deployed on the cluster whenever the source code in its GitHub repository is updated.
+
+.Prerequisites
+
+* {pipelines-title} must be installed on your cluster.
+
+* You have installed the OpenShift CLI (`oc`).
+
+* You have installed the Knative CLI (`kn`).
+
+* Access to the GitHub repository containing your function's source code.
+
+.Procedure
+
+. Update the `func.yaml` file in your function project to include the GitHub repository URL and the branch to watch for updates:
++
+[source,yaml]
+----
+name: my-function
+namespace: ""
+runtime: node
+...
+repository: https://github.com/john-smith/my-repository.git <1>
+branch: main <2>
+----
+<1> Repository URL
+<2> Branch to monitor
+
+. To define the build configuration using Shipwright, create a `build.yaml` file in your project:
++
+[source,yaml]
+----
+apiVersion: shipwright.io/v1alpha1
+kind: Build
+metadata:
+  name: my-function-build
+spec:
+  source:
+    url: https://github.com/john-smith/my-repository.git <1>
+  strategy:
+    name: buildpacks-v3
+    kind: ClusterBuildStrategy
+  builder:
+    image: paketobuildpacks/builder:base
+  output:
+    image: image-registry.openshift-image-registry.svc:5000/my-project/my-function <2>
+----
+<1> Repository URL
+<2> Function image output
+
+. Apply the configuration to the {ocp-product-title} cluster:
++
+[source,terminal]
+----
+$ oc apply -f build.yaml
```

Comment:
this step fails with `error: unable to recognize "build.yaml": no matches for kind "Build" in version "shipwright.io/v1alpha1"`
Maybe it requires some operator pre-installed (i.e.: "Shipwright Operator"). I tried this one but it didn't work either.